src/utilstrencodings.cpp

```diff
@@ -14,17 +14,18 @@
 
 using namespace std;
 
-string SanitizeString(const string& str)
+static const string SAFE_CHARS[] =
 {
-    /**
-     * safeChars chosen to allow simple messages/URLs/email addresses, but avoid anything
-     * even possibly remotely dangerous like & or >
-     */
-    static string safeChars("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890 .,;_/:?@()");
-    string strResult;
+    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890 .,;_/:?@()", // SAFE_CHARS_DEFAULT
+    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890 .,;_?@" // SAFE_CHARS_UACOMMENT
+};
+
+string SanitizeString(const string& str, int rule)
```

Comment:
Instead of the slightly static rule parameter, would a charset represented as string with additional forbidden chars not be more flexible.
Instead `SanitizeString("Comment2 .,_?@", SAFE_CHARS_UACOMMENT)` it would then be `SanitizeString("Comment2 .,_?@", "/:()")`. 
Not sure if we wan't to have use case specific handling within `SanitizeString()`.